File: ollama/codigo-final/pidigits/codegemma-7b/cot/temp_0.7/pidigits.java
import java.math.BigInteger;

public class Pidigits {

    public static void main(String[] args) {
        int N = Integer.parseInt(args[0]);
        BigInteger q = BigInteger.ONE;
        BigInteger r = BigInteger.ZERO;
        BigInteger t = BigInteger.ONE;
        int k = 0;
        int count = 0;

        while (true) {
            next_step(q, r, t, k);
            int d = extract(3, q, r, t);
            if (d == extract(4, q, r, t)) {
                System.out.print(d);
                produce(d, q, r, t);
                count++;
                if (count % 10 == 0) {
                    System.out.print("\t:" + N);
                    if (count == N) {
                        break;
                    }
                    System.out.println();
                }
            }
        }
    }

    private static void next_step(BigInteger q, BigInteger r, BigInteger t, int k) {
        k++;
        r = r.multiply(BigInteger.TWO).add(q);
        r = r.multiply(BigInteger.valueOf(2 * k + 1));
        t = t.multiply(BigInteger.valueOf(2 * k + 1));
        q = q.multiply(BigInteger.valueOf(k));
    }

    private static int extract(int x, BigInteger q, BigInteger r, BigInteger t) {
        return (int) ((q.multiply(BigInteger.valueOf(x)).add(r)).divide(t)).longValue();
    }

    private static void produce(int d, BigInteger q, BigInteger r, BigInteger t) {
        q = q.multiply(BigInteger.TEN);
        r = r.subtract(t.multiply(BigInteger.valueOf(d))).multiply(BigInteger.TEN);
    }
}

File: ollama/codigo-final/pidigits/codegemma-7b/cot/temp_0.7/pidigits.cpp
#include <iostream>
#include <gmpxx.h>

using namespace std;

mpz_class q(1), r(0), t(1);
int k = 0;

void next_step() {
  k++;
  r = (2 * q + r) * (2 * k + 1);
  t *= (2 * k + 1);
  q *= k;
}

mpz_class extract(mpz_class x) {
  return (q * x + r) / t;
}

void produce(mpz_class d) {
  q *= 10;
  r = (r - t * d) * 10;
}

int main() {
  int N;
  cin >> N;

  while (true) {
    while (q > r) {
      mpz_class d1 = extract(3);
      mpz_class d2 = extract(4);
      if (d1 == d2) {
        cout << d1;
        produce(d1);
        k++;
        if (k == N) {
          cout << "\t:" << N << endl;
          return 0;
        }
      } else {
        next_step();
      }
    }
  }
}
